app/src/main/res/layout/fragment_keyboard.xml

```diff
@@ -94,14 +111,82 @@
         android:layout_marginStart="@dimen/keyboard_side_margin"
         android:layout_marginTop="@dimen/keyboard_keys_top_margin"
         android:layout_marginEnd="@dimen/keyboard_side_margin"
-        android:layout_marginBottom="@dimen/keyboard_keys_bottom_margin"
+        android:layout_marginBottom="@dimen/keyboard_side_margin"
         android:columnCount="10"
         android:rowCount="4"
         android:useDefaultMargins="true"
-        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintBottom_toTopOf="@+id/keyboard_bottom_bar"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/predictive_text" />
+        app:layout_constraintTop_toBottomOf="@+id/action_button_container" />
+
+    <LinearLayout
```

Comment:
Could this be a ConstraintLayout with `app:layout_constraintHorizontal_weight` specified instead of a weighted LinearLayout?